component/vdr/cheqd/vdr.go

```diff
@@ -0,0 +1,245 @@
+type VDR struct {
+	endpointURL      string
+	client           *http.Client
+	accept           Accept
+	resolveAuthToken string
+}
+
+// Accept is method to accept did method.
+type Accept func(method string) bool
+
+// Option configures the peer vdr.
+type Option func(opts *VDR)
+
+// New creates new DID Resolver.
+func New(endpointURL string, opts ...Option) (*VDR, error) {
+	v := &VDR{client: &http.Client{}, accept: func(method string) bool { return true }}
+
+	for _, opt := range opts {
+		opt(v)
+	}
+
+	// Validate host
+	_, err := url.ParseRequestURI(endpointURL)
+	if err != nil {
+		return nil, fmt.Errorf("base URL invalid: %w", err)
+	}
+
+	v.endpointURL = endpointURL
+
+	return v, nil
+}
+
+// WithAccept option is for accept did method.
+func WithAccept(accept Accept) Option {
+	return func(opts *VDR) {
+		opts.accept = accept
+	}
+}
+
+// Accept did method - attempt to resolve any method.
+func (v *VDR) Accept(method string) bool {
+	return v.accept(method)
+}
+
+// Update did doc.
+func (v *VDR) Update(didDoc *diddoc.Doc, opts ...vdrapi.DIDMethodOption) error {
+	return fmt.Errorf("not supported")
+}
+
+// Deactivate did doc.
+func (v *VDR) Deactivate(did string, opts ...vdrapi.DIDMethodOption) error {
+	return fmt.Errorf("not supported")
+}
+
+// Close method of the VDR interface.
+func (v *VDR) Close() error {
+	return nil
+}
+
+// Read resolves a did:cheqd did.
+func (v *VDR) Read(didID string) (*did.DocResolution, error) {
+	httpClient := &http.Client{}
+
+	address := parseDIDCheqd(v.endpointURL, didID)
+
+	req, err := http.NewRequest("GET", address, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("error GET request from %s", address)
+	}
+
+	req.Header.Add("Accept", "application/ld+json")
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:cheqd did --> http request unsuccessful --> %w", err)
+	}
+
+	defer closeResponseBody(resp.Body)
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("http server returned status code [%d]", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:cheqd did --> error reading https response body: %s --> %w", body, err)
+	}
+
+	rawDocCheqd := &RawDocCheqd{}
+	if err = json.Unmarshal(body, &rawDocCheqd); err != nil {
+		return nil, fmt.Errorf("error unmarshal did:cheqd did --> error unmarshal https response body: %s --> %w", body, err)
+	}
+
+	rawDoc := convertTypeRawDocCheqdToRawDoc(rawDocCheqd)
+
+	body, err = json.Marshal(rawDoc)
+	if err != nil {
+		return nil, fmt.Errorf("error marshal did:cheqd did --> error marshaling https response body: %s --> %w", body, err)
+	}
+
+	doc, err := did.ParseDocument(body)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:cheqd did --> error parsing did doc --> %w", err)
+	}
+
+	return &did.DocResolution{DIDDocument: doc}, nil
+}
+
+// RawDoc type.
+type rawDoc struct {
```

Comment:
Do we need to re-declare that interface? It looks like a common one from the aries-framework-go.

Reply:
Yep, It's correct)

Reply:
We have an existing structure `rawDoc` in `aries-framework-go`, but we can't use this structure because this structure `rawDoc` is declared with a small letter in another package. Because of this, I didn't use this structure in my code.

Link to structure `rawDoc` from aries-framework-go:
> https://github.com/hyperledger/aries-framework-go/blob/566dff81e2ab98c0e33ffa4cbcee1bb926857c4f/pkg/doc/did/doc.go#L428

Reply:
Why don't you use a `Doc` structure: https://github.com/hyperledger/aries-framework-go/blob/566dff81e2ab98c0e33ffa4cbcee1bb926857c4f/pkg/doc/did/doc.go#L263?

Reply:
Yes, I can.

Reply:
Done! ✅